### PR TITLE
[WIP] website: Add service mesh page

### PR DESCRIPTION
**Pull request by roaks3**
**Note:** Final PR will be against the upstream repository

First pass at the modifications for consul.io Service Mesh page. I generally leaned towards less work, and existing styles, since the designs are still being worked on. Once those are finalized, we can do another pass and get things the way we want.

This should contain all (or almost all) of the content needed for this page.

### CODE REVIEW

File: website/source/mesh.html.erb [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+          <div>
+            <h3>Observability</h3>
+            <p>Enable networking metric collection, distributed tracking and logging via proxies with centralized configuration to provide insights into application behavior and performance without code modifications</p>
+          </div>
+        </div>
+        <div>
+          <picture>
+             <source type="image/png" srcset="
+              /assets/images/consul-connect/observability.png" />
+              <img src='/assets/images/consul-connect/observability.png' alt='Observability'>
```
Comment (roaks3) [inline website/source/mesh.html.erb]: This new png should be changed to a jpg, and preferably optimized the same way as some of the other ui-* images

**Comment by roaks3** [comment]
Update: Images have been optimized, and content/layout should all be good here. Remaining items are:
- Update navigation to include Service Mesh page
- Remove any old pages that are no longer needed with this change

File: website/source/index.html.erb [hunk 2]
```diff
@@ -112,32 +112,22 @@ description: |-
       <div class='g-use-cases'>
         <div>
           <div>
-            <img src='/assets/images/consul-connect/svgs/discovery-simple.svg' alt='Service Discovery'>
-            <h3>Service Discovery <span>for connectivity</h3>
-            <p>Service Registry enables services to register and discover each other.</p>
+            <img src='/assets/images/consul-connect/discovery-simple.png' alt='Service Discovery'>
```
Comment (roaks3) [inline website/source/index.html.erb]: TODO (but maybe not for this PR): This image, and the one below for the mesh, should really be svgs instead of pngs

**Comment by roaks3** [comment]
This should be ready for review now. Essentially adds `mesh.html`, and all required assets, and removes `configuration.html` and `segmentation.html`, and all of their unique assets.

**Comment by rjspiker** [comment]
@roaks3 Looks great to me.  The only thing I'm seeing is the bleeding on these images when in a mobile view:

Screen Shot 2019-06-07 at 3 16 18 PM

**Pull request closed without merging**